main_test.go

```diff
@@ -80,59 +104,63 @@ func TestEndpoints(t *testing.T) {
+		{"/search?prerelease=true", "/search", "search-package-prerelease.json", defaultSearchHandler},
+		{"/search?prerelease=foo", "/search", "search-package-prerelease-error.txt", defaultSearchHandler},
+		{"/search?category=datastore&prerelease=true", "/search", "search-category-datastore-prerelease.json", defaultSearchHandler},
+		{"/search?type=content&prerelease=true", "/search", "search-content-packages.json", defaultSearchHandler},
+		{"/search?type=input&prerelease=true", "/search", "search-input-packages.json", defaultSearchHandler},
+		{"/search?type=input&package=integration_input&prerelease=true", "/search", "search-input-integration-package.json", defaultSearchHandler},
+		{"/search?type=integration&package=integration_input&prerelease=true", "/search", "search-integration-integration-package.json", defaultSearchHandler},
+		{"/search?capabilities=observability,security&prerelease=true", "/search", "search-prerelease-capabilities-observability-security.json", defaultSearchHandler},
+		{"/search?capabilities=none&prerelease=true", "/search", "search-prerelease-capabilities-none.json", defaultSearchHandler},
+		{"/search?spec.min=1.1&spec.max=2.10&prerelease=true", "/search", "search-spec-min-1.1.0-max-2.10.0.json", defaultSearchHandler},
+		{"/search?spec.max=2.10&prerelease=true", "/search", "search-spec-max-2.10.0.json", defaultSearchHandler},
+		{"/search?spec.max=2.10.1&prerelease=true", "/search", "search-spec-max-error.txt", defaultSearchHandler},
+		{"/search?prerelease=true&discovery=fields:process.pid", "/search", "search-discovery-fields-process-pid.txt", defaultSearchHandler},
+		{"/search?prerelease=true&discovery=fields:non.existing.field", "/search", "search-discovery-fields-empty.txt", defaultSearchHandler},
+		{"/search?prerelease=true&discovery=datasets:good_content.errors", "/search", "search-discovery-datasets.txt", defaultSearchHandler},
+		{"/search?prerelease=true&discovery=datasets:good_content.errors&discovery=fields:process.pid", "/search", "search-discovery-multiple.txt", defaultSearchHandler},
+		{"/search?prerelease=true&discovery=datasets:good_content.errors&discovery=fields:process.path", "/search", "search-discovery-multiple-no-match.txt", defaultSearchHandler},
 		{"/favicon.ico", "", "favicon.ico", faviconHandleFunc},
 
 		// Removed flags, kept to ensure that they don't break requests from old versions.
-		{"/search?internal=true", "/search", "search-package-internal.json", searchHandler(testLogger, indexer, testCacheTime)},
+		{"/search?internal=true", "/search", "search-package-internal.json", defaultSearchHandler},
+
+		// Test queries with unknown query parameters
+		{"/search?package=yamlpipeline&unknown=true", "/search", "search-unknown-query-parameter-error.txt", defaultSearchHandler},
+		{"/search?package=yamlpipeline&unknown=true", "/search", "search-allowed-unknown-query-parameter.json", allowUnknownQueryParamsSearchHandler},
```

Comment:
Added a few tests to ensure that works as expected depending on the value of `allowUnknownQueryParameters`